### PR TITLE
feat: add validations for records search filters and sort

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR add some changes related with how we search for records:
* Add a new `SearchRecordsQueryValidator` class that allow us to validate some details about a schema `SearchRecordsQuery`.
* Integrate this validation class into handlers doing searching for records. Returning a `422` error if this validation is not successful.
* Changed `gte` and `lte` attributes on `RangeFilter` to `ge` and `le`. The change is inspired on a previous use of `ge` and `le` for a Metadata schema so we don't want to add a new way to specify these kind of ranges.
* Add a new `RecordFilterScope` to be used on the query `sort` attribute like this:
```json
{
  "sort": [
    {"scope": {"entity": "record", "property": "inserted_at"}, "order": "desc"},
    {"scope": {"entity": "record", "property": "updated_at"}, "order": "desc"},
  ]
}
```
* I didn't add any validation to avoid `RecordFilterScope` to be used on `filters` attribute too so it should be ignored on this section for the search engine /cc @frascuchon
* Some more small changes, for example the addition of `get_question_by_name_and_dataset_id_or_raise` and `get_metadata_property_by_name_and_dataset_id_or_raise` to be used on `SearchRecordsQueryValidator` but that could be useful also once we fix current exception handling with FastAPI.

Refs #4227 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

- [x] Running unit tests locally.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by jfcalvo** [comment]
@damianpumar you can use this PR to check that your request for searching records are well constructed. Note that with these changes you will still not get results with new filters and sort applied (we are still not executing search engine queries with these parameters).

File: tests/unit/server/api/v1/datasets/test_search_dataset_records.py [hunk 1]
```diff
@@ -242,3 +244,49 @@ async def test_with_record_without_vector(self, async_client: AsyncClient, owner
...
+            json={
+                "query": {
+                    "text": {"q": "text", "field": "non-existent"},
+                },
+                "filters": {
+                    "and": [
+                        {
+                            "type": "terms",
+                            "scope": {"entity": "response", "question": "non-existent"},
+                            "values": ["value-a"],
```
Comment (frascuchon) [inline tests/unit/server/api/v1/datasets/test_search_dataset_records.py:263]: Maybe makes sense to parameterize this as having a wrong example per scope type (response, suggestion, and metadata property)

Comment (frascuchon) [reply tests/unit/server/api/v1/datasets/test_search_dataset_records.py:263]: Okay. I saw the specific unit tests for the validator. Forget this

**Review by frascuchon — APPROVED** [review]
Just move the  `errors.py` module under the `argilla.server` module, since is a server module.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**